core/src/main/scala/org/apache/spark/storage/StorageUtils.scala

```diff
@@ -224,8 +299,14 @@ class StorageStatus(val blockManagerId: BlockManagerId, val maxMem: Long) {
         } else {
           _rddStorageInfo(rddId) = (newMem, newDisk, newTachyon, level)
         }
+      case BroadcastBlockId(broadcastId, _) =>
+        if (newMem + newDisk + newTachyon == 0) {
+          _broadcastStorageInfo.remove(broadcastId)
+        } else {
+          _broadcastStorageInfo(broadcastId) = (newMem, newDisk, newTachyon)
+        }
```

Comment:
I think you could use the same `case rddOrBroadcastBlockId @ (_: BroadcastBlockId | _: RDDBlockId)` pattern here to combine both matches in this method ... its subjective whether its worth it or not, but I think it would be cleaner.